src/library_syscall.js

```diff
@@ -1009,6 +1009,8 @@ var SyscallsLibrary = {
     assert(!flags);
 #endif
     if (old.fd === newfd) return -{{{ cDefs.EINVAL }}};
+    // Check newfd is within range of valid open file descriptors.
+    if (newfd < 0 || newfd > 1024) return -{{{ cDefs.EBADF }}};
     var existing = FS.getStream(newfd);
```

Comment:
Actually, shouldn't this line error if `newfd` is invalid? That would be a more precise check than the maximum id of an fd.

Reply:
`FS.getStream()` returns undefined is the stream is not currently open its not supposed to throw an error.    For that we have `FS.getStreamChecked()` but we don't want to use that here since we want to allows for the case where newfs is not open.

I think this is a reasonable place to do that check since this is the only place where the user can supply "proposed" file descriptor that is not already open (AFAIK).